frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts

```diff
@@ -0,0 +1,87 @@
+import { kea } from 'kea'
+import { bulkInviteLogicType } from './bulkInviteLogicType'
+import { OrganizationInviteType } from '~/types'
+import api from 'lib/api'
+import { toast } from 'react-toastify'
+
+interface InviteType {
+    email: string
+    first_name?: string
+    isValid: boolean
+}
+
+interface BulkInviteResponse {
+    invites: OrganizationInviteType[]
+}
+
+const DEFAULT_INVITE = { email: '', first_name: '', isValid: true }
+const DEFAULT_INVITES = [DEFAULT_INVITE, DEFAULT_INVITE, DEFAULT_INVITE]
+
+export const bulkInviteLogic = kea<bulkInviteLogicType<BulkInviteResponse, InviteType>>({
+    actions: {
+        updateInviteAtIndex: (payload, index: number) => ({ payload, index }),
+        addMoreInvites: true,
+        resetInvites: true,
+    },
+    reducers: {
+        invites: [
+            DEFAULT_INVITES as InviteType[],
+            {
+                updateInviteAtIndex: (state, { payload, index }) => {
+                    const newState = [...state]
+                    newState[index] = { ...state[index], ...payload }
+                    return newState
+                },
+                addMoreInvites: (state) => {
+                    return [...state, DEFAULT_INVITE, DEFAULT_INVITE]
+                },
+                resetInvites: () => DEFAULT_INVITES,
+            },
+        ],
+    },
+    selectors: {
+        canSubmit: [
+            (selectors) => [selectors.invites],
+            (invites: InviteType[]) => {
+                let atLeastOneValidEmail = false
+                for (const invite of invites) {
+                    if (!invite.isValid) {
+                        return false
+                    }
+                    if (invite.email && invite.isValid) {
+                        atLeastOneValidEmail = true
+                    }
+                }
+                return atLeastOneValidEmail
```

Comment:
This can be simplified:

```ts
invites.filter((invite) => invite.isValid && !!invite.email).length > 0
```

Reply:
Well not quite, there are two things we check here,
a) at least one valid invite **and**,
b) no invalid invites

The proposed code doesn't cover case b).

Reply:
When is isValid true? Is it enough to check by that?

```
invites.length > 0 && invites.filter(({ isValid }) => !isValid).length == 0
```

Reply:
We default to `isValid = true`, let me make some adjustments so we can indeed use the simplified version here.